File: tauri/src-tauri/src/models/thought_queue.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ThoughtItem {
    pub id: String,
    pub workspace_id: String,
    pub content: String,
    pub status: String,
    pub process_at: Option<String>,
    pub model_name: String,
    pub prompt_prefix: String,
    pub result: Option<String>,
    pub result_at: Option<String>,
    pub session_id: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateThoughtRequest {
    pub workspace_id: String,
    pub content: String,
    pub process_at: Option<String>,
    pub model_name: Option<String>,
    pub prompt_prefix: Option<String>,
    pub session_id: Option<String>,
}

File: swift/Sources/Aetherium/Models/ThoughtQueue.swift
import Foundation
import SwiftData

@Model
final class ThoughtQueueItem {
    @Attribute(.unique) var id: UUID
    var content: String
    var status: String // ThoughtStatus.rawValue
    var processAt: Date?
    var modelName: String
    var promptPrefix: String
    var result: String?
    var resultAt: Date?
    var createdAt: Date
    var updatedAt: Date

    // Relationships
    var workspace: Workspace?

    init(
        id: UUID = UUID(),
        content: String,
        status: ThoughtStatus = .pending,
        processAt: Date? = nil,
        modelName: String = "",
        promptPrefix: String = ""
    ) {
        self.id = id
        self.content = content
        self.status = status.rawValue
        self.processAt = processAt
        self.modelName = modelName
        self.promptPrefix = promptPrefix
        self.createdAt = Date()
        self.updatedAt = Date()
    }

    var currentStatus: ThoughtStatus {
        ThoughtStatus(rawValue: status) ?? .pending
    }
}

enum ThoughtStatus: String, Codable {
    case pending
    case scheduled
    case processing
    case done
}
